src/modules/common.ts

```diff
@@ -1,35 +1,39 @@
 import BaseModel from '../models/baseModel'
+import PostModel from '../models/post'
+import PostsModel from '../models/posts'
+
+import produce from 'immer'
 
 export const startLoading = (state: BaseModel) => {
-	return {
-		...state,
-		loading: true,
-		error: undefined,
-	}
+	return produce(state, draft => {
+		draft.loading = true
+		draft.error = undefined
+	})
 }
 
 export const endLoading = (state: BaseModel, error) => {
-	return {
-		...state,
-		loading: false,
-		error,
-	}
+	return produce(state, draft => {
+		draft.loading = false
+		draft.error = error
+	})
 }
 
-export const updateData = (state: BaseModel, data) => {
-	return {
-		...state,
-		loading: false,
-		error: undefined,
-		...data,
-	}
+export const updateData = (state: PostModel, data) => {
```

Comment:
'common' is used for all models, not for PostModel only. I suggest not to implement immer here. Instead do it in reducer and pass the draft to these common functions